packages/components/src/lib/controllers/settings-controller.js

```diff
@@ -0,0 +1,264 @@
+    this._resourceOptions = null
+    this._uiOptions = null
+
+    /*
+    A copy of resource options for the lookup UI component
+    This doesn't get reloaded from the storage adapter because
+    we don't expose it to the user via preferences
+     */
+    this._lookupResourceOptions = null
+    this._siteOptions = null
+
+    /**
+     * Holds a shard app-wide API object
+     *
+     * @type {object}
+     * @private
+     */
+    this._api = {}
+
+    // A shared Vuex store object
+    this._store = null
+
+    // An object that stores config settings of a controller
+    this._config = {
+    }
+  }
+
+  async init ({
+    api, store, configServiceUrl, clientId, appName, appVersion, branch, buildNumber, storageAdapter,
+    featureOptionsDefaults, resourceOptionsDefaults, uiOptionsDefaults, siteOptionsDefaults
+  } = {}) {
+    if (!api) {
+      throw new Error('API object is required for a settings controller initialization')
+    }
+    if (!store) {
+      throw new Error('Vuex store is required for a settings controller initialization')
+    }
+
+    this._api = api
+    this._store = store
+    this._storageAdapter = storageAdapter
+
+    this._configServiceUrl = configServiceUrl
+    this._clientId = clientId
+    this._appName = appName
+    this._appVersion = appVersion
+    this._branch = branch
+    this._buildNumber = buildNumber
+
+    this._featureOptionsDefaults = featureOptionsDefaults
+    this._resourceOptionsDefaults = resourceOptionsDefaults
+    this._uiOptionsDefaults = uiOptionsDefaults
+    this._siteOptionsDefaults = siteOptionsDefaults
+
+    const appConfigPromise = this.requestAppOptions()
+    const initOptionsPromise = this.initOptions()
+    /*
+    We do not allow user to configure lookupResourceOptions or siteOptions currently.
+    Because of this, they are initialized separately from the rest of the options.
+    `initOptions()` will be called several times during the app's lifetime,
+    after a user has been logged in, as one example.
+    `initNonConfigurableOptions()` will be called only once,
+    during an initialization of the Settings Controller.
+     */
+    this.initNonConfigurableOptions()
+    const [appConfigResponse] = await Promise.all([appConfigPromise, ...initOptionsPromise])
+    this._appConfig = await appConfigResponse.json() // Parse an app config's response into JSON
+
+    // region Public API of a settings controller
+    // A public API must be defined before modules are created because modules may use it
+    this._api.settings = {
+      initOptions: this.initOptions.bind(this),
+      getLexisOptions: () => this._appConfig && this._appConfig['lexis-cs'] ? this._appConfig['lexis-cs'] : {},
+      // TODO: logeion options are currently under the `lexis-cs` branch, at least in a dev config. Shall we fix it?
+      getLogeionOptions: () => this._appConfig && this._appConfig['lexis-cs'] && this._appConfig['lexis-cs'].logeion ? this._appConfig['lexis-cs'].logeion : {},
```

Comment:
that is not right. In the master branch they are retrieved from 'logeion'. From input-autocomplete.vue:

```   fetchOptions: this.app.config.logeion```

So if the config file is wrong we should fix it and the code should definitely retrieve them from the logeion branch of the config not lexis-cs.

Reply:
What I've meant is that when I want to retrieve configuration (I'm using the following URL: https://config.alpheios.net/v1/config?clientId=embed-lib-index-demo&appName=Alpheios%20Embedded%20Library&appVersion=3.3.0&buildBranch=dev-ui-controller-refactoring&buildNumber=20200714700) I'm getting the following back:
```
{
  "lexis-cs": {
      "cedict":{
           "target_url":"https://lexis-cs.alpheios.net/index.html"
       },
      "logeion":{
           "baseurl":"https://api-v2.logeion.org/logeion2/search","apikey":"AIzaSyBDCq6R_QkHldUpiEvUt3GyReIBhvyj9NI"
       }
    }
}
```
`logeion` is within the `lexis-cs` branch. I think this is not right because they're not related. Shouldn't it be something like:
```
{
  "lexis-cs": {
      "cedict":{
           "target_url":"https://lexis-cs.alpheios.net/index.html"
       },
    },
   "logeion":{
      "baseurl":"https://api-v2.logeion.org/logeion2/search","apikey":"AIzaSyBDCq6R_QkHldUpiEvUt3GyReIBhvyj9NI"
   }
}
```
instead? What do you think?

Reply:
yes I agree. I think this is a problem with the config file however. The code should reference it in the "logeion" branch as it is in the master branch of alpheios-core. I'll enter an issue to fix the config file.

Reply:
https://github.com/alpheios-project/alpheios-config-api/issues/4

Reply:
OK, I will update the code to refer to the the root `logeion` branch in the remote config.